src/utils/getSupportedChainIdFromWalletConnectSession.ts

```diff
@@ -0,0 +1,35 @@
+import type { SessionTypes } from '@walletconnect/types'
+import { SupportedChainId } from 'constants/chains'
+
+// Helper function to extract chainId from string in format 'eip155:{chainId}'
+function getChainIdFromFormattedString(item: string): number | null {
+  const splitItem = item.split(':')
+  return splitItem.length > 1 && !isNaN(Number(splitItem[1])) ? Number(splitItem[1]) : null
+}
+
+export function getSupportedChainIdFromWalletConnectSession(session?: SessionTypes.Struct): SupportedChainId[] {
+  if (!session?.namespaces) return []
+
+  const eip155Keys = Object.keys(session.namespaces)
+  const namespaces = Object.values(session.namespaces)
+
+  // Collect all arrays into one for unified processing
+  const allItems = [
```

Comment:
technically wouldn't a "bad" namespace request like this with mismatched chains and accounts pass this function and return [10, 137, 1, 42161]?

not a huge deal since that should be an issue on the wallet side 

```
 const namespaces = {
      eip155: {
        chains: ['10', '137'],
        accounts: ['eip155:1:0xc0ffee254729296a45a3885639AC7E10F9d54979', 'eip155:42161:address'],
        methods: [],
        events: [],
      },
    }
```

Reply:
Yeah it would pass, but I don't really want to go down the route of enforcing the spec client side 🙈